pennylane_lightning/src/util/Util.hpp

```diff
@@ -312,8 +312,10 @@ inline auto transposed_state_index(size_t ind,
                                    const std::vector<size_t> &new_axes)
     -> size_t {
     size_t new_index = 0;
-    for (size_t axis : new_axes) {
-        new_index += (ind % 2) << axis;
+    const size_t max_axis = new_axes.size() - 1;
+    // NOLINTNEXTLINE(modernize-loop-convert)
```

Comment:
Why do we not want this modernized?

Reply:
Because the proposed change uses `std::ranges::reverse_view`, and this is not working fine on MacOS.
Also Tidy will not be happy with his own suggestion, saying that:
`no viable constructor or deduction guide for deduction of template arguments of 'reverse_view' [clang-diagnostic-error]`
You can see more here:
https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/3245244628/jobs/5322519450

Reply:
Yes. I think it makes sense.